cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/model/impl/ResultSetTupleSnapshot.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.model.impl;
+
+import com.datastax.driver.core.ColumnDefinitions;
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.Row;
+
+import org.hibernate.ogm.model.spi.TupleSnapshot;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Wrap a java-driver Row, handling column name and type mappings.
+ *
+ * @author Jonathan Halliday
+ */
+public class ResultSetTupleSnapshot implements TupleSnapshot {
+
+	private final Row row;
+	private Map<String, Integer> columnNames = new HashMap<String, Integer>();
+	private ProtocolVersion protocolVersion;
```

Comment:
It's not a constant, it's runtime depending on the cluster you're connecting to. If it's not here then it's in the dialect and all the tuples need a backpointer to that instead. Really a design flaw in the Row/DataType - it should know how to deserialize itself, but can't really cope with dynamic typing. It's app friendly, not framework friendly. /sigh.  We'll just have to live with it unless we can get a change pushed upstream.